lib/spot-checker.js

```diff
@@ -98,26 +111,26 @@ export default class SpotChecker {
         clearTimeout(timeout)
       }
       timeout = setTimeout(() => {
-        stats.timeout = true
+        retrievalStats.timeout = true
         controller.abort()
       }, 60_000)
     }
 
     try {
-      const url = getRetrievalUrl({ protocol, address, cid })
+      const url = getRetrievalUrl({ protocol, address, cid, dagScope })
       console.log(`Fetching: ${url}`)
 
       resetTimeout()
       const res = await this.#fetch(url, { signal })
-      stats.statusCode = res.status
+      retrievalStats.statusCode = res.status
 
       if (res.ok) {
         resetTimeout()
         const reader = await CarBlockIterator.fromIterable(res.body)
         for await (const block of reader) {
-          stats.byteLength += block.bytes.length
+          retrievalStats.byteLength += block.bytes.length
           await validateBlock(block)
-          if (maxByteLength > 0 && stats.byteLength > maxByteLength) {
+          if (maxByteLength > 0 && retrievalStats.byteLength > maxByteLength) {
```

Comment:
```suggestion
          if (maxByteLength !=== undefined && retrievalStats.byteLength > maxByteLength) {
```

Maybe I am missing something and it's important to preserve the current behaviour of `maxByteLength=0`, WDYT?